### PR TITLE
OnBlur function and a way to get all values from a collection.

### PR DESCRIPTION
**Pull request by danielronnkvist**
I saw that there was a PR already with the feature to create tags on blur.

The `getTagValues` function on the `JSTagsCollection` was necessary for my project where I had several _js-tag_ fields and wanted to use the values for each field. Perhaps `getNextTag` and `getPreviousTag` could be used instead of my solution but I thought it was a to complicated way to solve it.

Checks if the added tag isn't an empty string.

Also trimmed trailing white spaces.

### CODE REVIEW

File: jsTag/compiled/jsTag.debug.js [hunk 1]
```diff
@@ -332,6 +341,13 @@ jsTag.factory('InputService', ['$filter', function($filter) {
     }
   }
 
+
+  InputService.prototype.onBlur = function(tagsCollection) {
+    if (this.input !== "") {
```
Comment (eranhirs) [inline jsTag/compiled/jsTag.debug.js]: The if clause is unnecessary

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "jsTag",
-  "version": "0.3.5",
+  "version": "0.3.6",
```
Comment (eranhirs) [inline package.json]: Please use version 0.3.5.1

File: bower.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "jsTag",
-  "version": "0.3.5",
+  "version": "0.3.6",
```
Comment (eranhirs) [inline bower.json]: Please use version 0.3.5.1

**Comment by eranhirs** [comment]
Thank you for contributing!
- I added some line notes.
- The configuration info is great. Currently all Usage information is in the github pages, please move it to there.

File: jsTag/source/javascripts/services/inputService.js [hunk 4]
```diff
@@ -94,6 +101,11 @@ jsTag.factory('InputService', ['$filter', function($filter) {
 
       // Split value by spliter (usually ,)
       var values = originalValue.split(options.splitter);
+      for (var i = 0; i < values.length; i++) {
```
Comment (eranhirs) [inline jsTag/source/javascripts/services/inputService.js]: Please explain these lines

Comment (danielronnkvist) [inline jsTag/source/javascripts/services/inputService.js]: When a tag ends with a char from `options.splitter` the resulting array is `["tag", ""]` and an empty tag object is created. This is an attempt to delete false(i.e. `""`) values from the array.

File: jsTag/source/templates/default/js-tag.html [hunk 5]
```diff
@@ -37,6 +37,7 @@
     ng-hide="isThereAnEditedTag"
     ng-keydown="inputService.onKeydown(inputService, tagsCollection, {$event: $event})"
     placeholder="{{options.texts.inputPlaceHolder}}"
+    ng-blur="inputService.onBlur(tagsCollection)"
```
Comment (eranhirs) [inline jsTag/source/templates/default/js-tag.html]: Please add the same change to the typeahead template also (templates/typeahead/js-tag.html)

File: jsTag/source/javascripts/services/inputService.js (hunk 4 above)
Comment (eranhirs) [inline jsTag/source/javascripts/services/inputService.js]: Add a comment explaining that please

**Comment by eranhirs** [comment]
Looks great, added last line notes and let's merge this

**Pull request merged**